code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,243 @@
+		return
+
+	addtimer(CALLBACK(src, .proc/crash_shuttle, target), 30 SECONDS)
+
+/datum/game_mode/crash/setup()
+	SSjob.DivideOccupations() 
+	create_characters() //Create player characters
+	collect_minds()
+	reset_squads()
+
+	// Reassign everyone to the same squad
+	for(var/i in GLOB.new_player_list)
+		var/mob/new_player/N = i
+		var/mob/living/carbon/human/player = N.new_character
+		if(!istype(player) || !player?.mind.assigned_role)
+			continue
+
+		player.change_squad("Alpha") // TODO: There could be a better way to do this, but we need to rework squads / jobs / etc.
+
+	equip_characters()
+
+	transfer_characters()	//transfer keys to the new mobs
+
+	return TRUE
+
+/datum/game_mode/crash/pre_setup()
+	. = ..()
+	// Spawn the ship
+	if(!SSmapping.shuttle_templates[shuttle_id])
+		message_admins("Gamemode: Shuttle [shuttle_id] wasn't found and can't be loaded")
+		CRASH("Shuttle [shuttle_id] wasn't found and can't be loaded")
+		return FALSE
+
+	var/datum/map_template/shuttle/S = SSmapping.shuttle_templates[shuttle_id]
+	var/obj/docking_port/stationary/L = SSshuttle.getDock("crashmodeloading")
+	shuttle = SSshuttle.action_load(S, L)
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	// Create xenos
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_queen(M)
+		else
+			transform_xeno(M)
+
+
+#define CRASH_DRAW (1 << 0)
+#define CRASH_XENO_MAJOR (1 << 1)
+#define CRASH_XENO_MINOR (1 << 2)
+#define CRASH_MARINE_MINOR (1 << 3)
+#define CRASH_MARINE_MAJOR (1 << 4)
+
+
+/datum/game_mode/crash/check_finished()
+	if(round_finished)
+		return TRUE
+
+	if(world.time < (SSticker.round_start_time + 15 SECONDS))
+		return FALSE
+
+	var/list/living_player_list = count_humans_and_xenos()
+	var/num_humans = living_player_list[1]
+	var/num_xenos = living_player_list[2]
+	
+	var/list/grounded_living_player_list = count_humans_and_xenos(SSmapping.levels_by_any_trait(list(ZTRAIT_GROUND)))
+	var/num_grounded_humans = grounded_living_player_list[1]
+	
+
+	var/victory_options = (num_humans == 0 && num_xenos == 0 || (planet_nuked && !marines_evac)) 		<< 0 // Draw, for all other reasons
+	victory_options |= (!planet_nuked && num_humans == 0 && num_xenos > 0) 								<< 1 // XENO Major (All marines killed)
+	victory_options |= (marines_evac && !planet_nuked) 													<< 2 // XENO Minor (Marines evac'd )
+	victory_options |= (marines_evac && planet_nuked && (num_humans == 0 || num_grounded_humans > 0)) 	<< 3 // Marine minor (Planet nuked, some human left on planet)
+	victory_options |= (marines_evac && planet_nuked && num_grounded_humans == 0) 						<< 4 // Marine Major (Planet nuked, marines evac)
+	switch(victory_options)
```

Comment:
The switch won't work well with the flags, as the result could be mixed.

Reply:
I dont think thats the case here, this was suggested by Ninja originally to improve readability.

Reply:
If you have an example though, im happy to update it

Reply:
I like it tbh

Reply:
Added a todo to fix the nuke win condition, but otherwise this is good as is.